roles/app_server/tasks/main.yml

```diff
@@ -1,135 +0,0 @@
-
-- name: Install unicorn config
-  template: src=unicorn.rb.j2
-            dest={{app_path}}/config/unicorn.rb
-  tags: [configure_unicorn]
-
-- name: Set up db config
-  template: src=database.yml.j2 dest={{app_path}}/config/database.yml
-
-# zzet.rbenv puts all the rbenv stuff in profile for some reason
-# so we gotta use login shells to do this stuff
-- name: Install bundle gems
-  remote_user: "{{ deployer_user.name }}"
-  command: chdir={{ app_path }}
-           bash -lc "RAILS_ENV={{app_env}} bundle install
-             --without test development contests_development --deployment"
-  tags: [deploy, bundle]
-  when: (app_checkout is defined and app_checkout.changed) or
-        force_bundle
-
-- name: Ensure env_config.yml file present
-  stat: path={{ app_path }}/config/env_config.yml
-  register: env_config_file
-  tags: [deploy]
-
-- name: Ask for env_config.yml
-  debug: msg="You've got to upload env_config.yml to {{app_path}}/config to continue"
-  when: env_config_file.stat.exists != true
-  tags: [deploy]
-
-- name: Wait one day for env_config.yml to get put on the server
-  wait_for: path={{app_path}}/config/env_config.yml state=present timeout=86400
-  when: env_config_file.stat.exists != true
-  tags: [deploy]
-
-- name: Ensure secrets.yml file present
-  stat: path={{ app_path }}/config/secrets.yml
-  register: secrets_file
-  tags: [deploy]
-
-- name: Ask for secrets.yml
-  debug: msg="You've got to upload secrets.yml to {{app_path}}/config to continue"
-  when: secrets_file.stat.exists != true
-  tags: [deploy]
-
-- name: Wait one day for secrets.yml to get put on the server
-  wait_for: path={{app_path}}/config/secrets.yml state=present timeout=86400
-  when: secrets_file.stat.exists != true
-  tags: [deploy]
-
-- include: db_reset.yml
-  when: force_db_reset is defined and force_db_reset
-  tags: [db_reset]
-
-- name: Precompile Assets
-  remote_user: deployer
-  run_once: true
-  command: chdir={{ app_path }}
-           bash -lc "bundle exec rake assets:precompile RAILS_ENV={{app_env}}"
-  tags: [precompile_assets, deploy]
-  when: precompile_assets
-
-- name: Migrate DB
-  remote_user: deployer
-  command: chdir={{ app_path }}
-           bash -lc "bundle exec rake db:migrate RAILS_ENV={{app_env}}"
-  run_once: true
-  tags: [deploy, migrate, db_reset]
-  when: (app_checkout is defined and app_checkout.changed)
-        or rake.force_migrate 
-        or (db_reset is defined and db_reset.changed)
-
-- name: Seed DB
-  remote_user: deployer
-  command: chdir={{ app_path }}
-           bash -lc "bundle exec rake db:seed RAILS_ENV={{app_env}}"
-  run_once: true
-  tags: [deploy, seed, db_reset]
-  when: (app_checkout is defined and app_checkout.changed)
-        or rake.force_seed
-        or (db_reset is defined and db_reset.changed)
-
-- name: Ensure tmp dir present for unicorn pids
-  file: state=directory path={{app_path}}/tmp/unicorn owner=deployer
-  tags: [deploy]
-
-- name: reload unicorn
```

Comment:
I put this in the `load_database` role, too, since I'm not too sure I can go back and run tasks in the `unicorn` role again at this point.